compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/expressions/QueryExpressionTest.java

```diff
@@ -256,4 +256,20 @@ public void testIntermediateClauseStartRecovery() {
     public void testSelectClauseStartRecovery() {
         testFile("query-expr/query_expr_source_69.bal", "query-expr/query_expr_assert_69.json");
     }
+
+    @Test
+    public void testQueryActionOrExprWithClientRemoteMethodCallAction() {
+        testFile("query-expr/query_expr_source_71.bal", "query-expr/query_expr_assert_71.json");
+    }
+
+    @Test
+    public void testQueryActionOrExprWithAsyncSendAction() {
```

Comment:
Shall we move these tests under `// Valid syntax`? 
Only tests with syntax diagnostics go under recovery section.